src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindCommandResolver.java

```diff
@@ -38,20 +40,46 @@ public Class<FindCommand> getCommandClass() {
   @Override
   public Operation resolveCommand(CommandContext commandContext, FindCommand command) {
     List<DBFilterBase> filters = resolve(commandContext, command);
+    final FindCommand.Options options = command.options();
     int limit =
-        command.options() != null && command.options().limit() != null
-            ? command.options().limit()
-            : documentConfig.maxLimit();
+        options != null && options.limit() != null ? options.limit() : documentConfig.maxLimit();
     int pageSize = documentConfig.defaultPageSize();
     String pagingState = command.options() != null ? command.options().pagingState() : null;
+    final SortClause sortClause = command.sortClause();
+    List<FindOperation.OrderBy> orderBy = null;
+    int skip = 0, maxSortReadLimit = 0;
+    ReadType readType = ReadType.DOCUMENT;
+    // If sort request
+    if (sortClause != null && !sortClause.sortExpressions().isEmpty()) {
+      orderBy =
+          command.sortClause().sortExpressions().stream()
+              .map(
+                  sortExpression ->
+                      new FindOperation.OrderBy(sortExpression.path(), sortExpression.ascending()))
+              .collect(Collectors.toList());
+      skip =
```

Comment:
Should we limit maximum skip value that can be defined? So it cannot be used to increase number of documents read beyond maximum limit? (and basically validation combination of skip+limit not to exceed maximum)

Reply:
The maxSortReadLimit is set as limit in cql so the read will stop once the limit is hit.

Reply:
@tatu-at-datastax But if skip > maxSortReadLimit think we can short circuit and return. In this case does it has to be a error or empty data set response.

Reply:
Hmmh. Good question; I can see either case. Maybe worth asking @amorton on opinion.